lib/utils.js

```diff
@@ -5,9 +5,23 @@ export const protocolWhitelist = new Set(['https:', 'http:'])
 export const userAgent =
   'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36'
 
+/**
+ * @typedef {object} Options
+ * @property {string} [baseUrl]
+ * @property {unknown} [agent]
+ * @property {import('http').IncomingHttpHeaders} [headers]
+ * @property {unknown} [timeout]
+ */
```

Comment:
I'm still not 💯 sure what this object is.
It's not quite a request options object, it's not quite got options.
I modeled the properties which are visible here, but would be interested on additional insights on how to more accurately represent this.

Reply:
We'd prolly have to refactor to TS to have a better type

Reply:
That is an option.
I'm not sure it would solve this specific issue.
The TypeScript in JSDoc mode is able to fully capture the interface.
`tsc` is passing and `type-coverage` is reporting every single variable has a valid (not `any`) type.
We could even leave this as-is and it would (likely) be fine.

What's unclear is what the interface actually represents.
Maybe it is a custom interface internal to this app? Maybe it is a type from another library?
Types should ideally also serve as a form of documentation which can capture the intent of what should be supported.
The issue I'm running into, is I don't know what the original intent is/was.